src/components/search-bar/category-search-bar/search-bar-category-dropdown/search-bar-category-dropdown.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { ReactNode, useRef, useState } from 'react';
+import { DropDownButton } from '@/components/drop-down/drop-down-button/drop-down-button';
+import { DropDownOption, DropDownOptions } from '@/components/drop-down-options/drop-down-options';
+import { useFocusTrap } from '@/hooks/use-focus-trap';
+import { useEscapePress } from '@/hooks/use-key-press';
+import { useOutsideClick } from '@/hooks/use-outside-click';
+import './search-bar-category-dropdown.scss';
+
+interface SearchBarCategoryDropdownProps<T, S> {
+  categories: DropDownOption<T, S>[];
+  selectedCategory: DropDownOption<T, S>;
+  onCategorySelect: (category: DropDownOption<T, S>) => void;
+  onClick: () => void;
+}
+
+export function SearchBarCategoryDropdown<T extends string, S extends ReactNode>({
+  selectedCategory,
+  categories,
+  onCategorySelect,
+  onClick,
+}: SearchBarCategoryDropdownProps<T, S>) {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const categoryDropdownRef = useRef(null);
+  useOutsideClick(categoryDropdownRef, () => setIsOpen(false));
+  useFocusTrap(categoryDropdownRef, isOpen);
+  useEscapePress(() => setIsOpen(false));
```

Comment:
both `useOutsideClick` and `useEscapePress` should only be active when `isOpen` is true, so let's pass that in as the third and second arg respectively

Reply:
I ended up scrapping this file and adding a `buttonClassName` to the `Dropdown` because it is doing the exact same thing, and the dropdown has it right.